### PR TITLE
Entity Column queries

### PR DESCRIPTION
**Pull request by dmdashenkov**
Support for the `EntityQuery`.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: gcd/src/main/java/org/spine3/server/storage/datastore/DsRecordStorage.java [hunk 1]
```diff
@@ -189,7 +205,85 @@ public EntityRecord apply(@Nullable Entity input) {
...
+
+    private Predicate<Entity> buildMemoryPredicate(boolean handleIds,
+                                                   Set<I> ids) {
+        final Predicate<Entity> idPredicate = handleIds
+                                              ? new IdFilter(ids)
+                                              : Predicates.<Entity>alwaysTrue();
+        return idPredicate;
+    }
+
+    @SuppressWarnings("unchecked") // Precise column type is undefined
```
Comment (armiol) [inline gcd/src/main/java/org/spine3/server/storage/datastore/DsRecordStorage.java]: `Exact column type is unknown at this stage.`

File: gcd/src/main/java/org/spine3/server/storage/datastore/DsRecordStorage.java [hunk 2]
```diff
@@ -189,7 +205,85 @@ public EntityRecord apply(@Nullable Entity input) {
...
+    private Filter buildColumnPredicate(Map<Column<?>, Object> columns) {
```
Comment (armiol) [inline gcd/src/main/java/org/spine3/server/storage/datastore/DsRecordStorage.java]: Please call the methods, which return `Filter`, accordingly. Right now they are called like they return `Predicate`.

File: gcd/src/main/java/org/spine3/server/storage/datastore/type/DsColumnTypes.java [hunk 3]
```diff
@@ -110,6 +117,11 @@ public void setColumnValue(BaseEntity.Builder storageRecord,
                                    String columnIdentifier) {
             storageRecord.set(columnIdentifier, value);
         }
+
+        @Override
+        public Value<?> toValue(String data) {
+            return StringValue.of(data);
```
Comment (armiol) [inline gcd/src/main/java/org/spine3/server/storage/datastore/type/DsColumnTypes.java]: This line is reported to be untested.

File: gcd/src/main/java/org/spine3/server/storage/datastore/type/DsColumnTypes.java [hunk 4]
```diff
@@ -132,6 +149,11 @@ public void setColumnValue(BaseEntity.Builder storageRecord,
                                    String columnIdentifier) {
             storageRecord.set(columnIdentifier, value);
         }
+
+        @Override
+        public Value<?> toValue(Long data) {
+            return LongValue.of(data);
```
Comment (armiol) [inline gcd/src/main/java/org/spine3/server/storage/datastore/type/DsColumnTypes.java]: This line is reported to be untested.

File: gcd/src/main/java/org/spine3/server/storage/datastore/type/DsColumnTypes.java [hunk 5]
```diff
@@ -143,6 +165,11 @@ public void setColumnValue(BaseEntity.Builder storageRecord,
                                    String columnIdentifier) {
             storageRecord.set(columnIdentifier, value);
         }
+
+        @Override
+        public Value<?> toValue(Boolean data) {
+            return BooleanValue.of(data);
```
Comment (armiol) [inline gcd/src/main/java/org/spine3/server/storage/datastore/type/DsColumnTypes.java]: Same as above.

File: gcd/src/main/java/org/spine3/server/storage/datastore/type/DsColumnTypes.java [hunk 6]
```diff
@@ -161,6 +188,11 @@ public void setColumnValue(BaseEntity.Builder storageRecord,
                                    String columnIdentifier) {
             storageRecord.set(columnIdentifier, value);
         }
+
+        @Override
+        public Value<?> toValue(DateTime data) {
+            return DateTimeValue.of(data);
```
Comment (armiol) [inline gcd/src/main/java/org/spine3/server/storage/datastore/type/DsColumnTypes.java]: Same as above.

File: gcd/src/main/java/org/spine3/server/storage/datastore/type/DsColumnTypes.java [hunk 7]
```diff
@@ -193,5 +230,10 @@ public void setColumnValue(BaseEntity.Builder storageRecord,
                                    String columnIdentifier) {
             storageRecord.set(columnIdentifier, value);
         }
+
+        @Override
+        public Value<?> toValue(String data) {
+            return StringValue.of(data);
```
Comment (armiol) [inline gcd/src/main/java/org/spine3/server/storage/datastore/type/DsColumnTypes.java]: Same as above.

File: gcd/src/main/java/org/spine3/server/storage/datastore/type/AbstractDatastoreColumnType.java [hunk 8]
```diff
@@ -44,4 +45,10 @@
     public void setNull(BaseEntity.Builder storageRecord, String columnIdentifier) {
         storageRecord.setNull(columnIdentifier);
     }
+
+    @Override
+    public void setColumnValue(BaseEntity.Builder storageRecord, C value, String columnIdentifier) {
+        final Value<?> dsValue = toValue(value);
```
Comment (armiol) [inline gcd/src/main/java/org/spine3/server/storage/datastore/type/AbstractDatastoreColumnType.java]: The body of this method is reported to be untested.

**Review by armiol — CHANGES_REQUESTED** [review]
@dmdashenkov LGTM in general. But the coverage does not seem to be thorough. See my comments on this matter; also please re-check yourself. 

The querying feature is very important, so the new code should be covered.

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

**Review by armiol — APPROVED** [review]
@dmdashenkov LGTM.

**Pull request merged**